dlt/common/data_writers/buffered.py

```diff
@@ -54,7 +56,7 @@ def write_data_item(self, item: TDataItems, columns: TTableSchemaColumns) -> Non
         if self._writer and not self._writer.data_format().supports_schema_changes and len(columns) != len(self._current_columns):
             self._rotate_file()
         # until the first chunk is written we can change the columns schema freely
-        self._current_columns = columns
+        self._current_columns = deepcopy(columns)
```

Comment:
your fix is correct as explained in the issue. I'm merging the PR and then will add another PR with tests. I will also change the `deepcopy` to shallow one, deepcopy is expensive and this function is often called

Reply:
Awesome! Thanks. Wasn't sure how the object gets mutated so `deepcopy` was the safer bet, better that it's not needed